model/lifecycle.go

```diff
@@ -54,6 +54,50 @@ type VersionToRestart struct {
 	TaskIds   []string `json:"task_ids"`
 }
 
+func UpdateVersionandPatchStatusForBuild(buildIds []string) error {
```

Comment:
nits: And should be capitalized, and should be Builds since we pass in multiple build Ids. 
Also, can you move this function to be next to https://github.com/evergreen-ci/evergreen/blob/5f8061f8f7315d12122d845241856244abaf3c76/model/task_lifecycle.go#L1051 since they are related?